Compilation error with intel compiler

With meep-1.19 Intel compiler stops with following:

--
CXX      sphere-quad.o
  CXXLD    sphere_quad
  CXX      array_slice.lo
In file included from meep.hpp(25),
                 from meep_internals.hpp(20),
                 from array_slice.cpp(27):
meep/vec.hpp(306): warning #3497: standard attribute syntax is a C++11 feature
  [[noreturn]] extern void abort(const char *, ...); // mympi.cpp
  ^

In file included from meep_internals.hpp(20),
                 from array_slice.cpp(27):
meep.hpp(818): error: qualified name is not allowed
    std::unique_ptr<binary_partition> bp;
    ^

In file included from meep_internals.hpp(20),
                 from  #array_slice.cpp(27):
meep.hpp(818): error #77: this declaration has no storage class or type specifier
    std::unique_ptr<binary_partition> bp;
    ^

In file included from meep_internals.hpp(20),
                 from array_slice.cpp(27):
meep.hpp(818): error: expected a ";"
    std::unique_ptr<binary_partition> bp;
                   ^

In file included from meep_internals.hpp(20),
                 from array_slice.cpp(27):
meep.hpp(822): error: namespace "std" has no member "unique_ptr"
  std::unique_ptr<binary_partition> choose_chunkdivision(grid_volume &gv, volume &v, int num_chunks,
       ^
In file included from meep_internals.hpp(20),
                 from array_slice.cpp(27):
meep.hpp(822): error: expected a ";"
  std::unique_ptr<binary_partition> choose_chunkdivision(grid_volume &gv, volume &v, int num_chunks,
                 ^

In file included from meep_internals.hpp(20),
                 from array_slice.cpp(27):
meep.hpp(1407): warning #3495: rvalue references are a C++11 feature
    void add_source(field_type ft, src_vol &&src);
                                           ^

In file included from meep_internals.hpp(20),
                 from array_slice.cpp(27):
meep.hpp(2183): error: namespace "std" has no member "unique_ptr"
    binary_partition(const split_plane &_split_plane, std::unique_ptr<binary_partition> &&left_tree,
                                                           ^

In file included from meep_internals.hpp(20),
                 from array_slice.cpp(27):
meep.hpp(2183): error: expected a ")"
    binary_partition(const split_plane &_split_plane, std::unique_ptr<binary_partition> &&left_tree,
                                                                     ^

In file included from meep_internals.hpp(20),
                 from array_slice.cpp(27):
meep.hpp(2200): error: qualified name is not allowed
    std::unique_ptr<binary_partition> left;
    ^

In file included from meep_internals.hpp(20),
                 from array_slice.cpp(27):
meep.hpp(2200): error #77: this declaration has no storage class or type specifier
    std::unique_ptr<binary_partition> left;
    ^
--

This error was inintroduced after meep-1.18.
I tried versions intel-2017.2.174  and intel-2019.0.117 with -std=c++11 option.

The switch "-std=c++11" was definitely added to CFLAGS and CXXFLAGS before running configure script. After digging I have found that CXXFLAGS inside Makefiles was set to just "-O3", probably from mpic++ installation (mvapich2). On editing Makefile inside src subdirectory, the compilation of meep 1.19 stopped here:

--
libtool: compile:  mpic++ -DHAVE_CONFIG_H -I. -I.. -I../src -I/home/Safe/panov/intel/include/ -I/home/Safe/panov/irus17/include/ -I/home/Safe/panov/include/ -march=core-avx2 -mtune=broadwell -O3 -std=c++11 -fPIC -finline-functions -fno-fast-math -I/home/Safe/panov/intel/include -MT structure_dump.lo -MD -MP -MF .deps/structure_dump.Tpo -c structure_dump.cpp  -fPIC -DPIC -o .libs/structure_dump.o
icpc: command line warning #10006: ignoring unknown option '-fno-fast-math'
structure_dump.cpp(427): error: no suitable conversion function from "const meep::split_plane" to "meep::direction" exists
  binary_partition::binary_partition(const binary_partition& other) : proc_id{other.proc_id}, plane{other.plane} {
                                                                                                    ^

compilation aborted for structure_dump.cpp (code 2)
make[2]: *** [structure_dump.lo] Error 1

--

--- a/src/structure_dump.cpp
+++ b/src/structure_dump.cpp
@@ -424,7 +424,7 @@ binary_partition::binary_partition(const split_plane &_split_plane,
   if (!left || !right) { meep::abort("Binary partition tree is required to be full"); }
 }
 
-binary_partition::binary_partition(const binary_partition& other) : proc_id{other.proc_id}, plane(other.plane) {
+binary_partition::binary_partition(const binary_partition& other) : proc_id(other.proc_id), plane(other.plane) {
   if (!other.is_leaf()) {
     left.reset(new binary_partition(*other.left));
     right.reset(new binary_partition(*other.right));

--- a/src/structure_dump.cpp
+++ b/src/structure_dump.cpp
@@ -424,7 +424,7 @@ binary_partition::binary_partition(const split_plane &_split_plane,
   if (!left || !right) { meep::abort("Binary partition tree is required to be full"); }
 }
 
-binary_partition::binary_partition(const binary_partition& other) : proc_id{other.proc_id}, plane{other.plane} {
+binary_partition::binary_partition(const binary_partition& other) : proc_id{other.proc_id}, plane(other.plane) {
   if (!other.is_leaf()) {
     left.reset(new binary_partition(*other.left));
     right.reset(new binary_partition(*other.right));

